symbolic-symcache/src/writer.rs

```diff
@@ -429,6 +443,22 @@ impl<'a> SymCacheConverter<'a> {
         if symbol.address as u32 >= *last_addr {
             self.last_addr = None;
         }
+
+        // Insert an explicit "empty" mapping for the end of the symbol.
+        // This is to ensure that addresses that fall "between" symbols don't get
+        // erroneously mapped to the previous symbol.
+        //
+        // We only do this if there is no previous mapping for the end address—we don't
+        // want to overwrite valid mappings.
+        //
+        // If the next symbol starts right at this symbols's end, that's no trouble,
+        // it will just overwrite this mapping.
+        if symbol.size > 0 {
+            let end_address = (symbol.address + symbol.size) as u32;
+            if let btree_map::Entry::Vacant(vacant_entry) = self.ranges.entry(end_address) {
+                vacant_entry.insert(NO_SOURCE_LOCATION);
+            }
+        }
```

Comment:
Out of curiosity, why do we need to add this logic in two different places?

Reply:
We insert both "symbol" and "functions" into the symcache. A symbol is extracted from the symbol in the debug file and has no line records, just a name and an address (and maybe a size). So functions and symbols take two separate code paths.